partner_manual_rank/models/res_partner.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2021 ForgeFlow, S.L.
+# Copyright 2022 Vauxoo, S.A.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+DOMAIN_SEARCH = {
+    ("=", False): ("=", 0),
+    ("=", True): (">=", 1),
+    ("!=", False): (">=", 1),
+    ("!=", True): ("=", 0),
+}
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    is_customer = fields.Boolean(
+        compute="_compute_is_customer",
+        inverse="_inverse_is_customer",
+        search="_search_is_customer",
+        string="Is a Customer",
+    )
+    is_supplier = fields.Boolean(
+        compute="_compute_is_supplier",
+        inverse="_inverse_is_supplier",
+        search="_search_is_supplier",
+        string="Is a Supplier",
+    )
```

Comment:
I prefered fields `store=True` as previous version https://github.com/OCA/partner-contact/blob/14.0/partner_manual_rank/models/res_partner.py#L13

Reply:
This field does not need to be stored, since the information it contains is identical to the information in the *`_rank` field, so it is redundant information. As it has the search method, it behaves identically to a stored field.

What did you think @luisg123v

Reply:
I think those fields should not be stored for two reasons:

- Since ranks are updated [via SQL](https://github.com/odoo/odoo/blob/120895356a5517ae47146a41539842f0d466256f/addons/account/models/partner.py#L560), such changes won't be identified by dependencies, so computed fields won't be updated.
- Even if they were updated, that would increase chances of concurrency issues. The reason Odoo moved such increasing to SQL is to avoid concurrency issues. But if the "customer" and "supplier" fields are updated as standard fields (by the ORM), the avoided concurrency will come back with those fields

Besides, as @frahikLV mentioned, almost all features are preserved as if the field were stored, as it's searchable.

Regards,

Reply:
thanks for your comments